subsys/bluetooth/controller/ll_sw/ull_scan_aux.c

```diff
@@ -93,32 +93,78 @@ void ull_scan_aux_setup(memq_link_t *link, struct node_rx_hdr *rx, uint8_t phy)
 	struct pdu_adv *pdu;
 	uint8_t aux_handle;
 	uint8_t *ptr;
+	uint8_t phy;
 
 	ftr = &rx->rx_ftr;
-	lll = ftr->param;
-	aux = (void *)HDR_LLL2EVT(lll);
-	if (((uint8_t *)aux < (uint8_t *)ll_scan_aux_pool) ||
-	    ((uint8_t *)aux > ((uint8_t *)ll_scan_aux_pool +
-			    (sizeof(struct ll_scan_aux_set) *
-			     (CONFIG_BT_CTLR_SCAN_AUX_SET - 1))))) {
-		/* NOTE: We are most probably ADV_EXT_IND on primary channel */
-		aux = NULL;
-	}
 
-	scan = NULL;
-	sync = NULL;
+	switch (rx->type) {
+	case NODE_RX_TYPE_EXT_1M_REPORT:
+		lll = NULL;
+		aux = NULL;
+		scan = (void *)HDR_LLL2EVT(ftr->param);
+		sync = scan->per_scan.sync;
+		phy = BT_HCI_LE_EXT_SCAN_PHY_1M;
+		break;
+	case NODE_RX_TYPE_EXT_CODED_REPORT:
+		lll = NULL;
+		aux = NULL;
+		scan = (void *)HDR_LLL2EVT(ftr->param);
+		sync = scan->per_scan.sync;
+		phy = BT_HCI_LE_EXT_SCAN_PHY_CODED;
+		break;
+	case NODE_RX_TYPE_EXT_AUX_REPORT:
+		lll = ftr->param;
+		aux = (void *)HDR_LLL2EVT(lll);
+		scan = (void *)HDR_LLL2EVT(aux->rx_head->rx_ftr.param);
+		sync = (void *)scan;
+		scan = ull_scan_is_valid_get(scan);
+		if (scan) {
+			sync = scan->per_scan.sync;
+			phy = lll->phy;
+			switch (phy) {
+			case BT_HCI_LE_EXT_SCAN_PHY_1M:
+				rx->type = NODE_RX_TYPE_EXT_1M_REPORT;
+				break;
+			case BT_HCI_LE_EXT_SCAN_PHY_2M:
+				rx->type = NODE_RX_TYPE_EXT_2M_REPORT;
+				break;
+			case BT_HCI_LE_EXT_SCAN_PHY_CODED:
+				rx->type = NODE_RX_TYPE_EXT_CODED_REPORT;
+				break;
+			default:
+				LL_ASSERT(0);
+				return;
+			}
+		} else {
+			rx->type = NODE_RX_TYPE_SYNC_REPORT;
+			rx->handle = ull_sync_handle_get(sync);
+
+			sync = NULL;
+			phy = lll->phy;
+		}
+		break;
 #if defined(CONFIG_BT_CTLR_SYNC_PERIODIC)
-	if (aux) {
-		struct lll_scan *lll_scan;
+	case NODE_RX_TYPE_SYNC_REPORT:
+		{
+			struct lll_sync *lll_sync;
 
-		lll_scan = aux->rx_head->rx_ftr.param;
-		scan = (void *)HDR_LLL2EVT(lll_scan);
-	} else {
-		scan = (void *)HDR_LLL2EVT(ftr->param);
-	}
+			lll_sync = ftr->param;
+			sync = (void *)HDR_LLL2EVT(lll_sync);
+
+			rx->handle = ull_sync_handle_get(sync);
```

Comment:
Please avoid using `sync` as temporary variable here and use a separate auto variable.

Reply:
Done